src/inmanta/data/__init__.py

```diff
@@ -4519,6 +4519,34 @@ async def get_resources_in_latest_version(
                         result.append(resource)
         return result
 
+    @classmethod
+    async def get_latest_resources_resource_type_count(cls, environment: uuid.UUID) -> JsonType:
```

Comment:
JsonType is a very generic type. It matches `Dict[str, Any]`. We can make this more specific `dict[str, int]` for this method.

```suggestion
    async def get_latest_resources_resource_type_count(cls, environment: uuid.UUID) -> dict[str, int]:
```